src/wham_v0.cpp

```diff
@@ -465,6 +591,847 @@ Type objective_function<Type>::operator() ()
+  for(int j = 0; j < n_growth_par; j++) { 
+	  for(int y = 0; y < n_years_model; y++) { 
+			for(int a = 0; a < n_ages; a++) { 
+				GW_par(y,a,j) = exp(growth_a(j,0) + growth_re(y,a,j)); 
+			}
+	  }
+   }
+
+  // add to growth parameters in projection years
+  if(do_proj == 1){ 
+  	for(int j = 0; j < n_growth_par; j++) { 
+	  if(proj_growth_opt(j) == 2){
+		  matrix<Type> GW_toavg(n_toavg,n_ages);
+		  for(int a = 0; a < n_ages; a++){
+			for(int i = 0; i < n_toavg; i++){
+			  GW_toavg(i,a) = GW_par(avg_years_ind(i),a,j);
+			}
+		  }
+		  vector<Type> GW_proj = GW_toavg.colwise().mean();
+		  for(int y = n_years_model; y < n_years_model + n_years_proj; y++){
+			for(int a = 0; a < n_ages; a++){
+			  GW_par(y,a,j) = GW_proj(a);
+			}
+		  }
+	  } else { // proj_GW_opt == 1
+		for(int y = n_years_model; y < n_years_model + n_years_proj; y++) {
+			for(int a = 0; a < n_ages; a++) { 		
+				GW_par(y,a,j) = exp(growth_a(j,0) + growth_re(y,a,j)); 
+			}
+		}
+	  }
+	}
+  }
+
+  // Construct LAA parameters per year
+  matrix<Type> LAA_par(n_years_model + n_years_proj,n_ages); 
+  for(int y = 0; y < n_years_model; y++) { 
+		for(int a = 0; a < n_ages; a++) { 
+			LAA_par(y,a) = exp(LAA_a(a) + LAA_re(y,a)); 
+		}
+  }
+
+
+  // add to LAA parameters in projection years
+  if((do_proj == 1) & (growth_model == 3)){ // LAA nonparametric
+	if(proj_LAA_opt(0) == 2){
+	  matrix<Type> GW_toavg(n_toavg,n_ages);
+	  for(int a = 0; a < n_ages; a++){
+		for(int i = 0; i < n_toavg; i++){
+		  GW_toavg(i,a) = LAA_par(avg_years_ind(i),a);
+		}
+	  }
+	  vector<Type> GW_proj = GW_toavg.colwise().mean();
+	  for(int y = n_years_model; y < n_years_model + n_years_proj; y++){
+		LAA_par.row(y) = GW_proj;
+	  }
+	} else { // proj_LAA_opt == 1
+		for(int y = n_years_model; y < n_years_model + n_years_proj; y++) {
+			for(int a = 0; a < n_ages; a++) { 		
+				LAA_par(y,a) = exp(LAA_a(a) + LAA_re(y,a)); 
+			}
+		}
+	}
+  }
+  
+  // add ecov effect on growth paramteres
+  int count_wheresub = 0; // used for growth and LW
+  for(int j = 0; j < n_growth_par; j++) { 
+	for(int i=0; i < n_Ecov; i++){
+		if((Ecov_where(i,n_effects-4) == 1) & ((Ecov_where_subindex(count_wheresub)-1) == j)) {  // for growth
+			for(int y = 0; y < n_years_model + n_years_proj; y++) {
+				for(int a = 0; a < n_ages; a++) { 
+					GW_par(y,a,j) *= exp(Ecov_lm(i,n_effects-4,y,0)); 
+				}
+			}
+			count_wheresub += 1; 
+		}
+	}
+  }
+
+  // add ecov effect on LAA
+	for(int i=0; i < n_Ecov; i++){
+		if((Ecov_where(i,n_effects-3) == 1)) { 
+			for(int y = 0; y < n_years_model + n_years_proj; y++) {
+				for(int a = 0; a < n_ages; a++) { 
+					LAA_par(y,a) *= exp(Ecov_lm(i,n_effects-3,y,a));
+				}
+			}
+		}
+	}
+	
+  // Update SD_par:
+  vector<Type> expSD(2); // always 2 parameters
+  for(int i=0; i < 2; i++) {
+	  if(growth_model < 3) expSD(i) = exp(SD_par(i)); 
+	  else expSD(i) = exp(SDLAA_par(i)); // nonparametric approach
+  }
+  REPORT(expSD);
```

Comment:
Rename expSD to something like growth_SD or something? Or drop it from the report?

Reply:
name changed. keep it reporting since it is important for output tables